src/cdomains/lval.ml

```diff
@@ -382,25 +388,38 @@ struct
 end
 
 (** Lvalue lattice with sublattice representatives for {!DisjointDomain}. *)
-module NormalLatRepr (Idx: IntDomain.Z) =
+module NormalLatRepr (Idx: IdxDomain) =
 struct
   include NormalLat (Idx)
 
+  module UnitIdxDomain =
+  struct
+    include Lattice.Unit
+    let equal_to _ _ = `Top
+    let to_int _ = None
+  end
   (** Representatives for lvalue sublattices as defined by {!NormalLat}. *)
   module R: DisjointDomain.Representative with type elt = t =
   struct
-    include Normal (Idx)
     type elt = t
 
-    let rec of_elt_offset: Offs.t -> Offs.t =
+    (* Offset module for representative without abstract values for index offsets, i.e. with unit index offsets.
+       Reason: The offset in the representative (used for buckets) should not depend on the integer domains,
+       since different integer domains may be active at different program points. *)
+    include Normal (UnitIdxDomain)
+
+    let rec of_elt_offset (*: (fieldinfo, Idx.t) offs -> (fieldinfo, UnitIdxDomain.t) offs *) =
       function
       | `NoOffset -> `NoOffset
       | `Field (f,o) -> `Field (f, of_elt_offset o)
-      | `Index (_,o) -> `Index (Idx.top (), of_elt_offset o) (* all indices to same bucket *)
-    let of_elt = function
+      | `Index (_,o) -> `Index (UnitIdxDomain.top (), of_elt_offset o) (* all indices to same bucket *)
+
+    let of_elt (x: elt): t = match x with
       | Addr (v, o) -> Addr (v, of_elt_offset o) (* addrs grouped by var and part of offset *)
       | StrPtr _ when GobConfig.get_bool "ana.base.limit-string-addresses" -> StrPtr None (* all strings together if limited *)
-      | a -> a (* everything else is kept separate, including strings if not limited *)
```

Comment:
We should keep this comment about the now-three remaining cases.

Reply:
I re-added the comment.